test/functional/lua/diagnostic_spec.lua

```diff
@@ -2444,6 +2444,36 @@ describe('vim.diagnostic', function()
       ]]
       )
     end)
+
+    it('works for multi-line diagnostics', function()
+      local result = exec_lua [[
+        local diagnostics = {
+          make_error("Syntax error", 0, 1, 1, 3),
+        }
+        local winids = {}
+        vim.api.nvim_win_set_buf(0, diagnostic_bufnr)
+        vim.diagnostic.set(diagnostic_ns, diagnostic_bufnr, diagnostics)
+        local _, w1 = vim.diagnostic.open_float(0, { header = false })
+        if w1 then
+          table.insert(winids, w1)
+          vim.api.nvim_win_close(w1, true)
+        end
+        vim.api.nvim_win_set_cursor(0, {2, 0})
+        local _, w2 = vim.diagnostic.open_float(0, { header = false })
+        if w2 then
+          table.insert(winids, w2)
+          vim.api.nvim_win_close(w2, true)
+        end
+        vim.api.nvim_win_set_cursor(0, {3, 0})
+        local _, w3 = vim.diagnostic.open_float(0, { header = false })
+        if w3 then
+          table.insert(winids, w3)
+          vim.api.nvim_win_close(w3, true)
+        end
+        return #winids
+      ]]
+      eq(2, result)
+    end)
```

Comment:
The fix looks right, but I'm not sure I follow what this test is doing? For one thing, I don't see any multi-line diagnostics. Why is the test result expecting 2 elements in `winids`? Can you either rewrite the test to be a little clearer, or at least add some clarifying comments to help understand the flow of the test?

Reply:
make_error the line is 0 row and 1 row mean line 1 and line 2. put cursor in line 1 has w1 in line 2 has w2 in line 3 nothing happen.

Reply:
updated and more clear now ❓

Reply:
Thanks for clarifying, I still think this test is needlessly obscure though. What does the result being `2` mean? You can use `exec_lua` and `eq` more than once in a single test, perhaps break the large `exec_lua` up into smaller bits, each of which tests some behavior (so it's clear exactly what's being tested).

Reply:
yup and done.